lib/vim-state.coffee

```diff
@@ -654,6 +662,16 @@ class VimState
     text = @getRegister(name)?.text
     @editor.insertText(text) if text?
 
+  ensureCursorIsWithinLine: (cursor) =>
+    return if @processing or @mode is 'visual' or @mode is 'insert'
```

Comment:
Can you explain a bit about why this `@processing` variable? Without looking into this too deeply, I would think that we could just `return unless @mode is 'normal'`.